core/node/node_sync/src/batch_status_updater/mod.rs

```diff
@@ -137,51 +509,121 @@ impl UpdaterCursor {
+                        L1TransactionValidationResult::TryLater => Ok((None, None, None)),
+                    }
+                } else {
+                    Ok((None, None, None))
+                }
+            }
+            AggregatedActionType::PublishProofOnchain => {
+                if let Some(prove_tx_hash) = batch_info.base.prove_tx_hash {
+                    if batch_info.base.prove_chain_id != Some(l1_transaction_verifier.sl_chain_id) {
+                        return Err(anyhow::anyhow!(
+                            "Prove transaction chain ID {:?} does not match SL chain ID {}",
+                            batch_info.base.prove_chain_id,
+                            l1_transaction_verifier.sl_chain_id
+                        ));
+                    }
+                    // Validate the prove transaction events.
+                    match l1_transaction_verifier
+                        .validate_prove_tx(prove_tx_hash, batch_info.number)
+                        .await
+                        .context("Failed to validate prove transaction")?
+                    {
+                        L1TransactionValidationResult::Valid => Ok((
+                            batch_info.base.prove_tx_hash,
+                            batch_info.base.proven_at,
+                            batch_info.base.prove_chain_id,
+                        )),
+                        L1TransactionValidationResult::TryLater => Ok((None, None, None)),
+                    }
+                } else {
+                    Ok((None, None, None))
+                }
+            }
+            AggregatedActionType::Execute => {
+                if let Some(execute_tx_hash) = batch_info.base.execute_tx_hash {
+                    if batch_info.base.execute_chain_id != Some(l1_transaction_verifier.sl_chain_id)
+                    {
+                        return Err(anyhow::anyhow!(
+                            "Execute transaction chain ID {:?} does not match SL chain ID {}",
+                            batch_info.base.execute_chain_id,
+                            l1_transaction_verifier.sl_chain_id
+                        ));
+                    }
+                    // Validate the execute transaction events.
+                    match l1_transaction_verifier
+                        .validate_execute_tx(execute_tx_hash, batch_info.number)
+                        .await
+                        .context("Failed to validate execute transaction")?
+                    {
+                        L1TransactionValidationResult::Valid => Ok((
+                            batch_info.base.execute_tx_hash,
+                            batch_info.base.executed_at,
+                            batch_info.base.execute_chain_id,
+                        )),
+                        L1TransactionValidationResult::TryLater => Ok((None, None, None)),
+                    }
+                } else {
+                    Ok((None, None, None))
+                }
```

Comment:
could we make a small macros for it?

Reply:
I created a function for this :)

Reply:
It's okish. You can simplify it further.